docs/website/docs/snippets/_iree-dev-packages.md

```diff
@@ -0,0 +1,17 @@
+=== ":octicons-git-branch-16: Development packages"
+
+    Development packages are built at every commit and on pull requests, for
+    limited configurations.
+
+    On **Linux** with **Python 3.11**, development packages can be installed
+    into a [Python `venv`](https://docs.python.org/3/library/venv.html) using
+    the
+    [`build_tools/pkgci/setup_venv.py`](https://github.com/iree-org/iree/blob/main/build_tools/pkgci/setup_venv.py)
+    script:
+
+    ``` shell
+    # Install packages from a specific commit ref.
+    # See also the `--fetch-latest-main` and `--fetch-gh-workflow` options.
+    python ./build_tools/pkgci/setup_venv.py /tmp/.venv --fetch-git-ref=8230f41d
+    source /tmp/.venv/bin/activate
+    ```
```

Comment:
I can also move this "development packages" snippet to its own pull request, since I expect other changes in this PR will be less contentious. Just let me know and I can split it up.

Reply:
I'd say it's fine to have as part of this and to iterate / improve our docs with future PRs, which we will do anyway :)